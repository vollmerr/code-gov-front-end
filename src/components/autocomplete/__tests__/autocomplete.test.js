import React from 'react';

import Autocomplete from 'components/autocomplete'
import CustomLink from 'components/custom-link'

import { shallow } from 'enzyme';

describe('Testing <Autocomplete /> component', () => {
  const props ={
    onClick: jest.fn(),
    options: [{text: 'test-text', to: "/test-route"}]
  }
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<Autocomplete {...props}/>);
  });
  it('renders Autocomplete with to', () => {
    expect(wrapper.exists()).toEqual(true);
  });
  it('tests onClick', () => {
    wrapper.find(CustomLink).simulate('click');
    expect(props.onClick).toBeCalled();
  });
  it('should render correct', () => {
    expect(wrapper).toMatchSnapshot();
  })
})