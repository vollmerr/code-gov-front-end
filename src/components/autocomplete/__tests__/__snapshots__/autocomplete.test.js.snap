// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <Autocomplete /> component should render correct 1`] = `
<ul
  className="autocomplete"
>
  <li
    key="/test-route"
  >
    <Connect(CustomLinkComponent)
      onClick={[Function]}
      to="/test-route"
    >
      <i
        className="icon icon-search"
      />
      <span>
        test-text
      </span>
    </Connect(CustomLinkComponent)>
  </li>
</ul>
`;
